x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -56,215 +58,272 @@ export class ManifestManager {
+    try {
+      let oldManifest: Manifest | null;
+
+      // Get the last-dispatched manifest
+      oldManifest = await this.getLastDispatchedManifest(ManifestConstants.SCHEMA_VERSION);
+
+      if (oldManifest === null && opts !== undefined && opts.initialize) {
+        oldManifest = new Manifest(
+          new Date(),
+          ManifestConstants.SCHEMA_VERSION,
+          ManifestConstants.INITIAL_VERSION
+        ); // create empty manifest
+      } else if (oldManifest == null) {
+        this.logger.debug('Manifest does not exist yet. Waiting...');
         return null;
-      });
-  }
+      }
 
-  public async refresh(opts?: ManifestRefreshOpts): Promise<WrappedManifest | null> {
-    let oldManifest: Manifest | null;
+      // Build new exception list artifacts
+      const artifacts = await this.buildExceptionListArtifacts(ArtifactConstants.SCHEMA_VERSION);
 
-    // Get the last-dispatched manifest
-    oldManifest = await this.getLastDispatchedManifest(ManifestConstants.SCHEMA_VERSION);
+      // Build new manifest
+      const newManifest = Manifest.fromArtifacts(
+        artifacts,
+        ManifestConstants.SCHEMA_VERSION,
+        oldManifest.getVersion()
+      );
 
-    if (oldManifest === null && opts !== undefined && opts.initialize) {
-      oldManifest = new Manifest(new Date(), ManifestConstants.SCHEMA_VERSION, 'v0'); // create empty manifest
-    } else if (oldManifest == null) {
-      this.logger.debug('Manifest does not exist yet. Waiting...');
+      // Get diffs
+      const diffs = newManifest.diff(oldManifest);
+
+      return {
+        manifest: newManifest,
+        diffs,
+      };
+    } catch (err) {
+      this.logger.error(err);
       return null;
     }
+  }
 
-    // Build new exception list artifacts
-    const artifacts = await this.buildExceptionListArtifacts(ArtifactConstants.SCHEMA_VERSION);
-
-    // Build new manifest
-    const newManifest = Manifest.fromArtifacts(
-      artifacts,
-      ManifestConstants.SCHEMA_VERSION,
-      oldManifest.getVersion()
-    );
-
-    // Get diffs
-    const diffs = newManifest.diff(oldManifest);
+  /**
+   * Syncs artifacts based on provided snapshot.
+   *
+   * Creates artifacts that do not yet exist and cleans up old artifacts that have been
+   * superceded by this snapshot.
+   *
+   * Can be filtered to apply one or both operations.
+   *
+   * @param snapshot
+   * @param diffType
+   */
+  public async syncArtifacts(snapshot: ManifestSnapshot, diffType?: 'add' | 'delete') {
+    const filteredDiffs = snapshot.diffs.reduce((diffs: ManifestDiff[], diff) => {
+      if (diff.type === diffType || diffType === undefined) {
+        diffs.push(diff);
+      } else if (!['add', 'delete'].includes(diff.type)) {
+        // TODO: replace with io-ts schema
+        throw new Error(`Unsupported diff type: ${diff.type}`);
+      }
+      return diffs;
+    }, []);
+
+    const adds = filteredDiffs.filter((diff) => {
```

Comment:
nit -> to be more readable it will be nice have branch logic to show that you are only doing add || delete on this function.